salt/sdb/env.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+'''
+Environment sdb Module
+
+:maintainer:    SaltStack
+:maturity:      New
+:depends:       None
+:platform:      all
+
+This module allows access to environment variables using an ``sdb://`` URI.
+
+Example configuration for this module:
+
+.. code-block:: yaml
+
+    osenv:
+      driver: env
+
+WARNING:
+--------
+OS environment variables will be available
+to read via SDB.
+Please make sure you don't have any sensitive data
+in your environment variables!!
+
+Example usage of sdb env module:
+
+.. code-block:: yaml
+
+    set some env var:
+      cmd.run:
+        - name: echo {{ salt['sdb.set']('sdb://osenv/foo', 'bar') }}
+        - order: 1
+
+    {% if salt['sdb.get']('sdb://osenv/foo') == 'bar' %}
+    always-changes-and-succeeds:
+      test.succeed_with_changes:
+        - name: foo
+    {% else %}
+    always-changes-and-fails:
+      test.fail_with_changes:
+        - name: foo
+    {% endif  %}
+
+The above example will return success.
+
+The ``env`` sdb module can also be used with salt cloud.
+Assuming you have exported the environment variable named
+``compute`` (and have ``osenv`` defined).
+The example below will look for the salt cloud config key ``compute_name``
+in the environment:
+
+.. code-block:: yaml
+
+  my-openstack-config:
+    compute_name: sdb://osenv/compute
+    ..snip
+
+'''
+from __future__ import absolute_import
+
+# import python libs
+from os import environ
+
+__func_alias__ = {
+    'set_': 'set'
+}
+
+
+def __virtual__():
+    '''
+    Always load
+    '''
+    return True
+
+
+def set_(key, value, profile=None):
```

Comment:
The set function used in the the above sls file will also cause a stacktrace when the profile kwarg is removed.

```python
'Rendering SLS \'base:test_sdb_env\' failed: Jinja error: set_() got an unexpected keyword argument \'profile\'
Traceback (most recent call last):
  File "/Users/adi/salt/salt/utils/templates.py", line 366, in render_jinja_tmpl
    output = template.render(**decoded_context)
  File "/Users/adi/envs/salt/lib/python2.7/site-packages/jinja2/environment.py", line 989, in render
    return self.environment.handle_exception(exc_info, True)
  File "/Users/adi/envs/salt/lib/python2.7/site-packages/jinja2/environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)\n  File "<template>", line 3, in top-level template code
  File "/Users/adi/salt/salt/modules/sdb.py", line 43, in set_
    return salt.utils.sdb.sdb_set(uri, value, __opts__)\n  File "/Users/adi/salt/salt/utils/sdb.py", line 65, in sdb_set
    return loaded_db[fun](query, value, profile=profile)\nTypeError: set_() got an unexpected keyword argument \'profile\';
 line 3\n\n---\nset some env var:\n  cmd.run:\n    - name: echo {{ salt[\'sdb.set\'](\'sdb://osenv/foo\', \'bar\') }}    <======================\n    - order: 1\n\n{% if salt[\'sdb.get\'](\'sdb://osenv/foo\') == \'bar\' %}\nalways-changes-and-succeeds:\n  test.succeed_with_changes:\n[...]\n---'
```